src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3468,3 +3445,355 @@ void c_typecheck_baset::make_constant_index(exprt &expr)
     throw 0;
   }
 }
+
+optionalt<symbol_exprt> c_typecheck_baset::typecheck_gcc_polymorphic_builtin(
```

Comment:
Consider moving this mega-function into a separate file and splitting into smaller ones

Reply:
Move to a separate file done, splitting up is still to-be-done.

Reply:
Refactoring completed.